GMX-Bots/gmx5-detect-frontruns/src/agent.ts

```diff
@@ -0,0 +1,206 @@
+import {
+  BlockEvent,
+  Finding,
+  HandleBlock,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  Log,
+  LogDescription,
+  ethers,
+  getEthersProvider,
+} from "forta-agent";
+import { NetworkManager } from "forta-agent-tools";
```

Comment:
Removed theProvider. As for the `networkManager`, I had it the way you suggested before, but had to change it because it was passed to `provideHandleTx` by value and not by reference, so the changes that happened in `initialize` didn't do anything. Lázaro suggested wrapping it in an object as a solution to this problem

Reply:
What you had before is the `router` as a parameter which will be passed by value as you said. 
My suggesting is passing the `networkManager` as a parameter to `initialize` and `provideHandleTx`.

Reply:
Okay, I understand what you mean now. The issue is if I have `networkManager` as a global variable, I can't assign it its value outside `initialize` or `provideHandleTx` (because there is an async call). So then I would have to change it's value when `initialize` is run. But wouldn't I run into the same issue, where `networkManager` will be passed to `provideHandleTx` by value?

Reply:
Yes, in that case you'll have the same problem. 

But you can avoid the async call when you create a new `NetworkManager` object, by passing only `util.data` as a parameter. Then initializing the chainId later by calling `init` function inside `initialize`.  You'll have something like: 

```
 const networkManager = new NetworkManager(util.data)

export const initialize = (provider: ethers.providers.Provider) => async () => {
await networkManager.init(provider)
...
}
```